test/test_transforms_v2_refactored.py

```diff
@@ -860,3 +862,448 @@ def test_transform_noop(self, input_type, device):
+
+        torch.manual_seed(seed)
+        params = transform._get_params([bounding_box])
+
+        torch.manual_seed(seed)
+        actual = transform(bounding_box)
+
+        expected = self._reference_affine_bounding_box(bounding_box, **params, center=center)
+
+        torch.testing.assert_close(actual, expected)
+
+    @pytest.mark.parametrize("degrees", _EXHAUSTIVE_TYPE_TRANSFORM_AFFINE_RANGES["degrees"])
+    @pytest.mark.parametrize("translate", _EXHAUSTIVE_TYPE_TRANSFORM_AFFINE_RANGES["translate"])
+    @pytest.mark.parametrize("scale", _EXHAUSTIVE_TYPE_TRANSFORM_AFFINE_RANGES["scale"])
+    @pytest.mark.parametrize("shear", _EXHAUSTIVE_TYPE_TRANSFORM_AFFINE_RANGES["shear"])
+    @pytest.mark.parametrize("seed", list(range(10)))
+    def test_transform_get_params_bounds(self, degrees, translate, scale, shear, seed):
+        image = self._make_input(torch.Tensor)
+        height, width = F.get_spatial_size(image)
+
+        transform = transforms.RandomAffine(degrees=degrees, translate=translate, scale=scale, shear=shear)
+
+        torch.manual_seed(seed)
+        params = transform._get_params([image])
+
+        if isinstance(degrees, (int, float)):
+            assert -degrees <= params["angle"] <= degrees
+        else:
+            assert degrees[0] <= params["angle"] <= degrees[1]
+
+        if translate is not None:
+            width_max = int(round(translate[0] * width))
+            height_max = int(round(translate[1] * height))
+            assert -width_max <= params["translate"][0] <= width_max
+            assert -height_max <= params["translate"][1] <= height_max
+        else:
+            assert params["translate"] == (0, 0)
+
+        if scale is not None:
+            assert scale[0] <= params["scale"] <= scale[1]
+        else:
+            assert params["scale"] == 1.0
+
+        if shear is not None:
+            if isinstance(shear, (int, float)):
+                assert -shear <= params["shear"][0] <= shear
+                assert params["shear"][1] == 0.0
+            elif len(shear) == 2:
+                assert shear[0] <= params["shear"][0] <= shear[1]
+                assert params["shear"][1] == 0.0
+            elif len(shear) == 4:
+                assert shear[0] <= params["shear"][0] <= shear[1]
+                assert shear[2] <= params["shear"][1] <= shear[3]
+        else:
+            assert params["shear"] == (0, 0)
+
+    @pytest.mark.parametrize("param", ["degrees", "translate", "scale", "shear", "center"])
+    @pytest.mark.parametrize("value", [0, [0], [0, 0, 0]])
+    def test_transform_sequence_len_errors(self, param, value):
+        if param in {"degrees", "shear"} and not isinstance(value, list):
+            return
+
+        kwargs = {param: value}
+        if param != "degrees":
+            kwargs["degrees"] = 0
+
+        with pytest.raises(
+            ValueError if isinstance(value, list) else TypeError, match=f"{param} should be a sequence of length 2"
```

Comment:
Probably not worth addressing in this PR because it's not related to tests, but do you have any thought on unifying or parameter error types to always be ValueError? Distinguishing between ValueError and TypeError doesn't seem worth it and leads to that kind of special-casing.

Reply:
I started out with the strong believe that one should differentiate between the two based on the [documentation](https://docs.python.org/3/library/exceptions.html#TypeError):

> Passing arguments of the wrong type (e.g. passing a [list](https://docs.python.org/3/library/stdtypes.html#list) when an [int](https://docs.python.org/3/library/functions.html#int) is expected) should result in a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError).

However, in practice it turns out that this distinction is not really helpful. I doubt anyone is doing

```python
try:
    torchvision.transforms.functional.affine(...)
except TypeError:
    # do something
except ValueError:
    # do something else
```

The most likely scenario is that no one is using the exceptions in their regular flow, but rather only if there is actual an error. And in that case a human is going to look into it and thus the message is much more important.

So overall I see few reasons to differentiate the two in `torchvision`. That might be different for other packages, where the users actually rely on the exception behavior. However, it doesn't bother me enough to actually go and address it. I'll review / stamp a PR if you want to bring one.

As a side note: the distinction between `TypeError` and `ValueError` here is terrible IMO, because we even use the same error message.